cmd/server/cadence/server.go

```diff
@@ -175,7 +175,7 @@ func (s *server) startService() common.Daemon {
 		}
 
 		params.ESConfig = advancedVisStore.ElasticSearch
-		esClient, err := elasticsearch.NewClient(params.ESConfig)
+		esClient, err := elasticsearch.NewGenericElasticSearchClient(params.ESConfig, s.cfg.Persistence.VisibilityConfig, params.Logger)
```

Comment:
Why we need a new `NewGenericElasticSearchClient`? My feeling is es7/8 just need to implement existing Client interface (https://github.com/uber/cadence/blob/6eb918ea60ab63c2f9896ef2d712abfe29a44619/common/elasticsearch/client.go#L34)

Reply:
yeah agreed. I plan to have that config in the next PR when introducing V7 as client.

Reply:
The client interface is not implementable by v7/v8. The v7/v8 versions of ElasticSearch are major version difference. They need to be run with a completely different library. The current Client interface is already binded with v6. For example, `*elastic.SearchResult` is the result just for v6.

Reply:
So the very  basic idea is that the interface needs to be defined in a way that it doesn't touch/refer to any v6 library struct/interfaces.

Reply:
Agreed about the config. I will add that in the next PR when introducing V7 as the implementation. This PR is just refactoring to make sure no behavior changes.